miden/tests/integration/stdlib/math/u64_mod.rs

```diff
@@ -880,6 +880,124 @@ fn unchecked_shr() {
     build_test!(source, &[5, a0, a1, b as u64]).expect_stack(&[c1, c0, 5]);
 }
 
+#[test]
+fn overflowing_shl() {
+    let source = "
+        use.std::math::u64
+        begin
+            exec.u64::overflowing_shl
+        end";
+
+    // shift by 0
+    let a: u64 = rand_value();
+    let (a1, a0) = split_u64(a);
+    let b: u32 = 0;
+
+    let c = (a as u128) * 2u128.pow(b);
```

Comment:
nit: I wonder if we should do something like `(a as u128) << b` here and in other such instances.